packages/xo-web/src/xo-app/vm/tab-console.js

```diff
@@ -91,9 +94,39 @@ export default class TabConsole extends Component {
     this.setState({ minimalLayout: !this.state.minimalLayout })
   }
 
+  _openSsh = () => {
+    window.open(`ssh://root@${this.props.vm.addresses['0/ip']}`)
+  }
+
+  _sshName = ({ value, onChange }) => (
+    <div>
+      <input
+        type='text'
+        className='form-control'
+        onChange={onChange}
+        value={value}
+      />
+    </div>
+  )
+
+  _openSshMore = async () => {
+    const cookieKey = `${this.props.vm.uuid}/ssh-user-name`
+    const suggestedUsername = cookies.get(cookieKey) || 'root'
+    const username = await form({
+      defaultValue: suggestedUsername,
+      header: <span>{_('sshUsernameLabel')}</span>,
+      render: props => <this._sshName {...props} />,
+    })
+    if (username !== suggestedUsername) {
```

Comment:
Is the test necessary?

Reply:
To avoid having a lot of useless cookies in the browser.

Reply:
Wdym? Past this `if`, there will always be 1 (and only 1) cookie for this VM, right?

Reply:
if the user changed their mind (or clicked the wrong button) and logged in with the default user, we will not create the cookie.